src/ConstantSum/ConstantSumSolver.sol

```diff
@@ -107,11 +115,50 @@ contract ConstantSumSolver {
         return encodeFeeUpdate(newSwapFee);
     }
 
+    /// @notice used by kit
     function prepareControllerUpdate(address newController)
         public
         pure
         returns (bytes memory)
     {
         return encodeControllerUpdate(newController);
     }
+
+    function getReservesAndLiquidity(uint256 poolId)
+        public
+        view
+        override
+        returns (uint256, uint256, uint256)
+    {
+        Pool memory pool = IDFMM(IStrategy(strategy).dfmm()).pools(poolId);
+        return (pool.reserves[0], pool.reserves[1], pool.totalLiquidity);
+    }
+
+    function getPoolParams(uint256 poolId)
+        public
+        view
+        returns (ConstantSumParams memory)
+    {
+        return abi.decode(
+            IStrategy(strategy).getPoolParams(poolId), (ConstantSumParams)
+        );
+    }
+
+    // These are same for allocation and deallocation
+    // delta y is 0
+    function PrepareAllocationDeltaGivenDeltaX(
+        uint256 poolId,
+        uint256 deltaX
+    ) public view returns (bytes memory) {
+        ConstantSumParams memory params = getPoolParams(poolId);
+        uint256 deltaL = deltaX.mulWadDown(params.price);
+        return abi.encode(deltaX, 0, deltaL);
+    }
+
+    function PrepareAllocationDeltaGivenDeltaY(
+        // uint256 poolId,
+        uint256 deltaY
+    ) public pure returns (bytes memory) {
+        return abi.encode(0, deltaY, deltaY);
+    }
```

Comment:
NatSpec could be improved and the name of the functions should use camel case.

Reply:
I changed some of the natspec but i am unsure if i am most familiar with the conventions you would like to adhere to.